controller/destination/listener.go

```diff
@@ -85,9 +86,13 @@ func (l *endpointListener) toWeightedAddrSet(endpoints []common.TcpAddress) *pb.
 			if len(resultingPods) == 0 || resultingPods[0] == nil {
 				log.Errorf("Could not find pod for IP [%s], this IP will be sent with no metric labels.", ipAsString)
 			} else {
-				pod := resultingPods[0]
-				metricLabelsForPod = pkgK8s.GetOwnerLabels(pod.ObjectMeta)
-				metricLabelsForPod["pod"] = pod.Name
+				for _, pod := range resultingPods {
+					if pod.Status.Phase == coreV1.PodRunning {
+						metricLabelsForPod = pkgK8s.GetOwnerLabels(pod.ObjectMeta)
+						metricLabelsForPod["pod"] = pod.Name
+						break
+					}
+				}
```

Comment:
tioli: if you split out a smaller method:

```go
func (l *endpointListener) toWeightedAddrSet(endpoints []common.TcpAddress) *pb.WeightedAddrSet {
 	addrs := make([]*pb.WeightedAddr, 0)
 	for i, address := range endpoints {
        a, err := l.toWeightedAddr(address)
        if err != nil {
          // ...
        }
		addrs = append(addrs, a)
	}

	return &pb.WeightedAddrSet{
		Addrs:        addrs,
		MetricLabels: l.labels,
    }
}

func (l *endpointListener) toWeightedAddr(address common.TcpAddress) (*pb.WeightedAddr, error) {
    wa := pb.WeightedAddr{
		Addr:         endpoints,
		Weight:       1,
		MetricLabels: map[string]string{},
     }

	ipAsString := util.IPToString(address.Ip)
	resultingPods, err := l.podsByIp.GetPodsByIndex(ipAsString)
	if err != nil {
        return err
	}

	if len(resultingPods) == 0 || resultingPods[0] == nil {
		log.Errorf("Could not find pod for IP [%s], this IP will be sent with no metric labels.", ipAsString)
        return wa, nil
	}

	for _, pod := range resultingPods {
		if pod.Status.Phase == coreV1.PodRunning {
			wa.MetricLabels = pkgK8s.GetOwnerLabels(pod.ObjectMeta)
			wa.MetricLabels["pod"] = pod.Name
			break
		}
	}
    return wa, nil
}
```

I find this to be slightly more readable, as the control flow in `toWeightedAddr` is easier for me to follow.